translate/src/modules/machinery/components/source/GoogleTranslation.tsx

```diff
@@ -1,27 +1,151 @@
-import React from 'react';
+import React, { useState, useRef, useContext, useEffect } from 'react';
 import { Localized } from '@fluent/react';
+import type { MachineryTranslation } from '~/api/machinery';
+import { fetchGPTTransform } from '~/api/machinery';
+import { Locale } from '~/context/Locale';
+import { logUXAction } from '~/api/uxaction';
+
+type Props = {
+  translation: MachineryTranslation;
+  onLLMTranslationChange: (llmTranslation: string) => void;
+};
 
 /**
  * Show the translation source from Google Translate.
  */
-export function GoogleTranslation(): React.ReactElement<'li'> {
+
+export function GoogleTranslation({
+  translation,
+  onLLMTranslationChange,
+}: Props): React.ReactElement<'li'> {
+  const [isDropdownOpen, setDropdownOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState('');
+  const [showOriginalOption, setShowOriginalOption] = useState(false);
+  const [currentTranslation, setCurrentTranslation] = useState(
+    translation.translation,
+  );
+  const dropdownRef = useRef<HTMLLIElement>(null);
+  const locale = useContext(Locale);
+  const [llmTranslation, setLLMTranslation] = useState('');
```

Comment:
The passing of the `llmTranslation` value up the stack is a bit of a code smell that there really ought to be a Context handling it, but at least it should be possible to avoid the `useEffect()` with a wrapper like this:

```suggestion
  const [llmTranslation, _setLLMTranslation] = useState('');
  const setLLMTranslation = (newTranslation: string) => {
    _setLLMTranslation(newTranslation);
    onLLMTranslationChange(newTranslation);
  }
```